python_modules/dagster/dagster/_core/definitions/asset_graph_subset.py

```diff
@@ -232,7 +232,7 @@ def from_storage_dict(
         cls,
         serialized_dict: Mapping[str, Any],
         asset_graph: AssetGraph,
-        allow_partial: bool = False,
```

Comment:
`from_storage_dict` is used in both asset backfill execution and loading backfill status.

In execution, we want to raise an error when partitions defs change so execution doesn't continue. For loading status, we can be forgiving and unpack partitions subsets even if the partitions defs have changed.